hdfs/util.py

```diff
@@ -22,12 +22,13 @@ class HdfsError(Exception):
 
   :param message: Error message.
   :param args: optional Message formatting arguments.
+  :param kwargs: optional extra values to store
 
   """
 
-  def __init__(self, message, *args):
+  def __init__(self, message, *args, **kwargs):
     super(HdfsError, self).__init__(message % args if args else message)
-
+    self.response = kwargs.get('response', None)
```

Comment:
Instead of special-casing this field in the base exception, I think your previous approach with an `HdfsStandbyError` was cleaner. Let's revert to it (see the related comment in the API handler too).